shoop/admin/modules/products/views/forms.py

```diff
@@ -263,3 +284,23 @@ def save(self, commit=True):
 class ProductImageMediaFormSet(ProductMediaFormSet):
     allowed_media_kinds = [ProductMediaKind.IMAGE]
     form_class = ProductImageMediaForm
+
+    def save(self, commit=True):
+        """
+        Add first saved image as  primary image for this product by default if none is selected as such
+        """
+        super(ProductImageMediaFormSet, self).save(commit)
+
+        if not self.forms:
+            return
+
+        has_primary = False
+        for form in self.forms:
+            if form.cleaned_data.get("is_primary"):
+                has_primary = True
+                break
```

Comment:
Too much detail for such a simple test. You could replace the lines 294-301 (from the `if not self.forms` to this line) with just
```python
has_primary = any(form.cleaned_data.get("is_primary") for form in self.forms)
```
Or if the `self.forms` could be None use `for form in (self.forms or [])` in that.